contracts/data/WitnetBoardData.sol

```diff
@@ -2,73 +2,111 @@
 
 pragma solidity >=0.7.0 <0.9.0;
 
-import "../libs/WitnetData.sol";
+import "../libs/Witnet.sol";
 
-/// @title Witnet Board base data model. 
+/// @title Witnet Request Board base data model. 
 /// @author The Witnet Foundation.
 abstract contract WitnetBoardData {  
 
-  struct WitnetBoardState {
-    address base;
-    address owner;    
-    uint256 numRecords;
-    mapping (uint => WitnetBoardDataRequest) requests;
-  }
-
-  struct WitnetBoardDataRequest {
-    WitnetData.Query query;
-    bytes result;
-  }
-
-  constructor() {
-    _state().owner = msg.sender;
-  }
-
-  modifier notDestroyed(uint256 _id) {
-    require(_id > 0 && _id <= _state().numRecords, "WitnetBoardData: not yet posted");
-    require(_getRequestQuery(_id).requestor != address(0), "WitnetBoardData: destroyed");
-    _;
-  }
-
-  modifier onlyOwner {
-    require(msg.sender == _state().owner, "WitnetBoardData: only owner");
-    _;    
-  }
-
-  modifier resultNotYetReported(uint256 _id) {
-    require(_getRequestQuery(_id).txhash == 0, "WitnetBoardData: already solved");
-    _;
-  }
-
-  modifier wasPosted(uint256 _id) {
-    require(_id > 0 && _id <= _state().numRecords, "WitnetBoardData: not yet posted");
-    _;
-  }
-
-  /// Gets admin/owner address.
-  function owner() public view returns (address) {
-    return _state().owner;
-  }
-
-  /// Returns storage pointer to contents of 'WitnetBoardState' struct.
-  function _state()
-    internal pure
-    returns (WitnetBoardState storage _ptr)
-  {
-    assembly {
-      _ptr.slot := _WITNET_BOARD_DATA_SLOTHASH
+    bytes32 internal constant _WITNET_BOARD_DATA_SLOTHASH =
+        /* keccak256("io.witnet.boards.data") */
+        0xf595240b351bc8f951c2f53b26f4e78c32cb62122cf76c19b7fdda7d4968e183;
+
+    struct WitnetBoardState {
+        address base;
+        address owner;    
+        uint256 numQueries;
+        mapping (uint => Witnet.Query) queries;
+    }
+
+    constructor() {
+        _state().owner = msg.sender;
+    }
+
+
+    modifier inStatus(uint256 _queryId, Witnet.QueryStatus _status) {
+      require(
+          _getQueryStatus(_queryId) == _status,
+          _getQueryStatusRevertMessage(_status)
+        );
+      _;
     }
-  }
-
-  /// Gets WitnetData.Query struct contents of given request.
-  function _getRequestQuery(uint256 _requestId)
-    internal view
-    returns (WitnetData.Query storage)
-  {
-    return _state().requests[_requestId].query;
-  }
-  
-  bytes32 internal constant _WITNET_BOARD_DATA_SLOTHASH =
-    /* keccak256("io.witnet.board.data") */
-    0x641d5bbf2c42118a382e660df7903a98dce7b5bb834d3ba9beae1890b2a72054;
+    modifier notDestroyed(uint256 _queryId) {
```

Comment:
I think it should be `notDeleted`, and it should be changed in WRBTrustableBase.sol aswell

Reply:
Right you are :-)